### DISCUSSION TITLE
[SQL Editor] How to disable addition extra spaces after autocompletion

### DISCUSSION THREAD

**Discussion by KES777** [dbeaver/dbeaver#11669] [Q&A] [answered]
#### System information: 
- Linux Mint 19.3
- DBeaver version 7.2.4

#### Connection specification:
- Database name and version
- Driver name
- Do you use tunnels or proxies (SSH, SOCKS, etc)? 

#### Describe the problem you're observing:
![image](https://user-images.githubusercontent.com/799021/99394254-42e82a80-28e7-11eb-9a5b-5f542393a29e.png)

While writing query and select from drop down list I got extra space which I delete every time.
In 6 version I did not face into this behaviour

#### Steps to reproduce, if exist:
I agree this is useful when I select field name and I at WHERE or at ON condition
But when I am at SELECT or chose tablename, the space is not required

**Comment by uslss** [comment]
hI, what database do you use?

**Comment by KES777** [comment]
PostgreSQL

**Comment by uslss** [comment]
could you provide a query example?

**Comment by KES777** [comment]
[dbeaver.mp4.zip](https://github.com/dbeaver/dbeaver/files/5583586/dbeaver.mp4.zip)
at where space is OK, at select space is not OK

**Comment by uslss** [comment]
unfortunately, the video isn't playable for me.
What formatter do you use? probably some special settings?
it will be awesome if you'll provide a query example, it will be very helpful fo issue investigation.
Thanks in advance

**Comment by KES777** [comment]
Video is created by kazam on linux. Probably you lack of some codecs =(.

Query example is on first post.
Sceenshots from video:

![image](https://user-images.githubusercontent.com/799021/102861572-90047280-4438-11eb-95de-fb9ce84e50a3.png)

![image](https://user-images.githubusercontent.com/799021/102861603-9b579e00-4438-11eb-907f-9a1c4511bf13.png)

![image](https://user-images.githubusercontent.com/799021/102861732-c4782e80-4438-11eb-876a-9184dc2e0492.png)

**Comment by kseniiaguzeeva** [comment, marked as answer]
Please try to open Properties and go to SQL Editor>Code completion and disable Insert space after table/column names. It should help.
![Screenshot 2020-12-22 at 10 46 57](https://user-images.githubusercontent.com/51405061/102863348-86343c80-4443-11eb-8001-dfaf8845acc0.png)

**Comment by KES777** [comment]
Thank you. But now I get vice-verca behaviour: at `WHERE` where space was useful I do not get it (not Ok :( )
at `SELECT` where space was unexpected. Now is OK

Space is useful feature but it require improvements

**Comment by kseniiaguzeeva** [comment]
Please give a query example for WHERE

**Comment by KES777** [comment]
```sql
select id, name                # <<<< example for select
from document d
where d.id = 33                # <<<< example for where
```

**Comment by LonwoLonwo** [comment]
Hello @KES777 
Could you explain - why do you need this improvement ("space just after WHERE")? The execution of the query will pass correctly in any case - with spaces and without.

**Comment by KES777** [comment]
because it looks ugly.

For example the one 40Kb query could be executed correct if we write it at one line. But still we split it to many lines for formatting purpose. Here is same it cause me to hit extra buttons to remove extra spaces at wrong place.

So if you implement feature it would be good to have it work properly.

**Comment by LonwoLonwo** [comment]
Formatting may look unpleasant for you, but be useful for someone else. We added this feature also under user request. As mentioned above, you can disable the setting in the preferences.

**Reply by givihuda** [reply]
It would be nice to have the setting for SELECT column names as well as a setting for other clauses like the WHERE clause... that's what OP was trying to describe. Right now the setting is just for table/column names ANYWHERE in the query.

It makes sense to have a space placed after the selected suggested column name in the WHERE clause (because no comma would be used there), but not in the SELECT (because when you specify multiple columns, they will be separated by a comma, but a space then a comma makes no syntactical or aesthetic sense).

Example:

```
SELECT column1 , column2 , column3
WHERE column1 = 5 AND column2 = 3
```

That's with the setting enabled:

![image](https://user-images.githubusercontent.com/74785256/125808506-f8dab11d-06ae-4b1e-b9ba-42f8d42fd7ec.png)

I've never seen any SQL having a space then a comma after each column name in the SELECT clause. So by default, DBeaver's column suggestions don't make sense with SELECT, even if they don't break query functionality. Yes it's an annoyance, but if the option is going to be there for a suggested column, why not have an option to not do it in SELECT?! Or at least not do it in SELECT when the option is enabled; this seems like natural behavior to have.

So, I've unchecked my option, because I more often have to specify multiple columns in SELECT than anywhere else. I usually write the queries in DBeaver just for testing purposes, so it's become too annoying to have to keep correcting that before copy-pasting code (which sometimes goes into code, but sometimes goes to other people, you know?).

Anyway, that's my two cents. I'm sure DBeaver understands the conundrum. I'd be surprised if people were specifying multiple columns less often in SELECT than anywhere else.
